suite-native/accounts/src/components/AddAccountsButton.tsx

```diff
@@ -10,28 +10,47 @@ import {
     StackNavigationProps,
 } from '@suite-native/navigation';
 import { IconButton } from '@suite-native/atoms';
-import { selectIsPortfolioTrackerDevice } from '@suite-common/wallet-core';
+import {
+    selectAreAllDevicesDisconnectedOrAccountless,
+    selectDeviceDiscovery,
+    selectIsPortfolioTrackerDevice,
+} from '@suite-common/wallet-core';
+
+import { useIsAddCoinAccountEnabled } from '../useIsAddCoinAccountEnabled';
 
 export const AddAccountButton = () => {
-    const isSelectedDevicePortfolioTracker = useSelector(selectIsPortfolioTrackerDevice);
     const navigation =
         useNavigation<StackNavigationProps<RootStackParamList, RootStackRoutes.AccountsImport>>();
 
+    const isSelectedDevicePortfolioTracker = useSelector(selectIsPortfolioTrackerDevice);
+    const discovery = useSelector(selectDeviceDiscovery);
+    const areAllDevicesDisconnectedOrAccountless = useSelector(
+        selectAreAllDevicesDisconnectedOrAccountless,
+    );
+    const { isAddCoinAccountEnabled } = useIsAddCoinAccountEnabled();
+
+    const shouldShowAddAccountButton =
+        isSelectedDevicePortfolioTracker ||
+        (isAddCoinAccountEnabled && !areAllDevicesDisconnectedOrAccountless && !discovery);
```

Comment:
The name of `selectAreAllDevicesDisconnectedOrAccountless` is very confusing for me, but I believe `!areAllDevicesDisconnectedOrAccountless` makes no sense in this condition because it can be `true` only for portfolio tracker and if selected device is portfolio tracker, this is always true (thanks to `isSelectedDevicePortfolioTracker ||`)